.devcontainer/setup.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/sh
+set -eux
+
+.devcontainer/docker-start.sh
+
+until docker info > /dev/null 2>&1 ; do
+  echo "Waiting for docker to start..."
+  sleep 0.1
+done
+
+echo "Setting up local container registry..."
+REGISTRY_NAME='registry.local'
+REGISTRY_PORT='5001'
+
+if [ "$(docker inspect -f '{{.State.Running}}' "${REGISTRY_NAME}" 2>/dev/null || true)" != 'true' ]; then
+  docker run \
+    -d --restart=always -p "127.0.0.1:${REGISTRY_PORT}:5000" --name "${REGISTRY_NAME}" \
+    registry:2
+fi
+
+echo "Creating KinD cluster..."
+cat <<EOF | kind create cluster --config=-
+
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+containerdConfigPatches:
+- |-
+  [plugins."io.containerd.grpc.v1.cri".registry.mirrors."localhost:${REGISTRY_PORT}"]
+    endpoint = ["http://${REGISTRY_NAME}:5000"]
+EOF
+
+echo "Connecting registry to KinD cluster..."
+if [ "$(docker inspect -f='{{json .NetworkSettings.Networks.kind}}' "${REGISTRY_NAME}")" = 'null' ]; then
+  docker network connect "kind" "${REGISTRY_NAME}"
+fi
+
+# Document the local registry
+# https://github.com/kubernetes/enhancements/tree/master/keps/sig-cluster-lifecycle/generic/1755-communicating-a-local-registry
+cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: local-registry-hosting
+  namespace: kube-public
+data:
+  localRegistryHosting.v1: |
+    host: "localhost:${REGISTRY_PORT}"
+    help: "https://kind.sigs.k8s.io/docs/user/local-registry/"
+EOF
+
+echo "Deploying cert-manager..."
+kubectl apply -f ./third_party/cert-manager-latest/cert-manager.yaml
+kubectl wait --for=condition=Established --all crd
+kubectl wait --for=condition=Available -n cert-manager --all deployments
+
+echo "Deploying Knative Serving..."
+ko apply --selector knative.dev/crd-install=true -Rf config/core/
+kubectl wait --for=condition=Established --all crd
+ko apply -Rf config/core/
+
+echo "Deploying Knative ingress with Kourier..."
+kubectl apply -f ./third_party/kourier-latest/kourier.yaml
+kubectl patch configmap/config-network \
+  -n knative-serving \
+  --type merge \
+  -p '{"data":{"ingress.class":"kourier.ingress.networking.knative.dev"}}'
```

Comment:
Have you tried to also set the DNS domain (e.g. with sslip.io to 127.0.0.1)? 
Also, we install [metal-lb](https://github.com/knative/serving/blob/main/.github/workflows/kind-e2e.yaml#L266) in our e2e-tests to get a routable service type `LoadBalancer` IP.
It would be great if this also works in Codespaces. Would you mind trying?

Reply:
Yep, installed `metal-lb`, added the load balancing logic (ie Setting up sslip.io) - things work fine. See https://github.com/knative/serving/pull/13838/commits/04d777067cf0d34f4f3dc8764ba379ae3e7b1020

Reply:
Nice, thank you!

Reply:
Can you or any other maintainer help approve this PR (provided it looks okay)? 
I'd appreciate that, thanks! :)